### PR TITLE
DS-675 | DS-676: update packages, install react-instantsearch

### PR DESCRIPTION
**Pull request by stfurkan**
# READY FOR REVIEW

# Summary
- DS-675: update packages
- DS-676: install `react-instantsearch`

# Review By (Date)
- End of sprint

# Criticality
- 4

# Review Tasks

## Setup tasks and/or behavior to test

1. Check out this branch
2. Review site and make sure everything works as expected
3. Review code

### CODE REVIEW

**Comment by ericbakenhus** [comment]
Login, general pages, travel study (an their APIs), and membership cards all seem good. 👍

@sherakama can you think of anything else that would be good to manually look at?

File: .eslintrc
Comment (sherakama) [inline .eslintrc:1]: Thanks for these updates.

File: cypress/support/e2e.ts [hunk 1]
```diff
@@ -1,3 +1,17 @@
 // import "gatsby-cypress/commands";
+import '@testing-library/cypress/add-commands';
 import './commands';
 import './setup';
+
+Cypress.on('uncaught:exception', (err) => {
+  // Cypress and React Hydrating the document don't get along
+  // for some unknown reason. Hopefully, we figure out why eventually
+  // so we can remove this.
```
Comment (sherakama) [inline cypress/support/e2e.ts:9]: Can you tell me more about this error or have links to bug reports?

Comment (stfurkan) [reply cypress/support/e2e.ts:9]: Here is one of the related topic: https://github.com/cypress-io/cypress/issues/27204

File: netlify.toml [hunk 2]
```diff
@@ -89,11 +90,3 @@
...
 [[plugins]]
   package = "@netlify/plugin-gatsby"
-
-# CONTEXTUAL PLUGINS
-# ##############################################################################
-[[context.production.plugins]]
-  package = "netlify-plugin-submit-sitemap"
-  [context.production.plugins.inputs]
-    # Path to the sitemap URL (optional, default = /sitemap.xml)
-    sitemapPath = "/sitemap/sitemap-index.xml"
```
Comment (sherakama) [inline netlify.toml:99]: Thanks for removing this.

File: package.json [hunk 3]
```diff
@@ -4,66 +4,69 @@
...
+    "query-string": "^9.0.0",
+    "react": "^18.0.0",
+    "react-dom": "^18.0.0",
     "react-helmet": "^6.1.0",
     "react-hero-icon": "^0.0.7",
-    "react-idle-timer": "^4.6.4",
-    "react-player": "^2.13.0",
-    "react-refresh": "^0.14.0",
+    "react-idle-timer": "^5.7.2",
+    "react-instantsearch": "^7.10.0",
```
[the remark this answers is not in the document]
Comment (stfurkan) [reply package.json:52]: Yes. Please see https://stanford.atlassian.net/browse/DS-676

[the remark this answers is not in the document]
Comment (sherakama) [reply package.json:52]: I did, it doesn't say much. Just double checking here.

File: cypress/support/e2e.ts (hunk 1 above)
Comment (sherakama) [reply cypress/support/e2e.ts:9]: Ty.

**Comment by sherakama** [comment]
Upgrade looks great @stfurkan. Nice work on this. 

It looks like auth and all the data is still flowing as it should and the pages are rendering as they were before. 

I found a new visual bug where the alert header is not showing up on the https://deploy-preview-850--stanford-alumni.netlify.app/membership/join/ page but it also exists on https://dev--stanford-alumni.netlify.app/membership/join so I will create a new ticket for that.

**Review by sherakama — APPROVED** [review]

**Comment by sherakama** [comment]
Screenshot 2024-06-04 at 12 29 58

**Comment by sherakama** [comment]
Screenshot 2024-06-04 at 12 31 37

**Review by ericbakenhus — APPROVED** [review]

**Pull request merged**